src/arche/rules/others.py

```diff
@@ -74,41 +75,46 @@ def fields_to_compare(source_df: pd.DataFrame, target_df: pd.DataFrame) -> bool:
     return False
 
 
-def garbage_symbols(items: Items) -> Result:
+def garbage_symbols(df: pd.DataFrame) -> Result:
     """Find unwanted symbols in `np.object` columns.
 
     Returns:
         A result containing item keys per field which contained any trash symbol
     """
     garbage = (
         r"(?P<spaces>^\s|\s$)"
-        r"|(?P<html_entities>&amp|&reg)"
-        r"|(?P<css>(?:(?:\.|#)[^#. ]+\s*){.+})"
-        r"|(?P<html_tags></?(h\d|b|u|i|div|ul|ol|li|table|tbody|th|tr|td|p|a|br|img|sup|SUP|"
+        r"|(?P<html_entities>&#?(?:\w*;|\d*;))"
+        r"|(?P<css>(?:\.|#|@)[^0-9{}#. ][^{}#.]+{(?:[^:;{}]+:[^:;{}]+;)+[\\n \n\t]*})"
```

Comment:
A few suggestions:
- Use a character set (`[…]`) for the first group of characters as well.
- Use `\s` instead of a literal space and `\d` instead of `0-9`.
- Make all matches [non-greedy](https://docs.python.org/3/howto/regex.html#greedy-versus-non-greedy).

```suggestion
        r"|(?P<css>[.#@][^\d{}#.\s][^{}#.]+?{(?:[^:;{}]+?:[^:;{}]+?;)+?[\\n \n\t]*?})"
```

And a question: what is `[\\n \n\t]` supposed to match? Currently it’s equivalent to `[n\n\t \\]`:

```
>>> re.findall(r'[\\n \n\t]', ' \n\t\\n\\tn\\t\\')
[' ', '\n', '\t', '\\', 'n', '\\', 'n', '\\', '\\']
```

Reply:
I guess it should be `\s*?`